draft-ietf-quic-http.md

```diff
@@ -1341,15 +1341,17 @@ type HTTP_LIMIT_EXCEEDED.
 
 The header of the request message is carried by a PUSH_PROMISE frame (see
 {{frame-push-promise}}) on the request stream which generated the push. This
-allows the server push to be associated with a client request. Ordering of a
-PUSH_PROMISE in relation to certain parts of the response is important (see
-Section 8.2.1 of {{!HTTP2}}).  Promised requests MUST conform to the
-requirements in Section 8.2 of {{!HTTP2}}.
+allows the server push to be associated with a client request.  Promised
+requests MUST conform to the requirements in Section 8.2 of {{!HTTP2}}.
 
 The same server push can be associated with additional client requests using a
-DUPLICATE_PUSH frame (see {{frame-duplicate-push}}).  Ordering of a
-DUPLICATE_PUSH in relation to certain parts of the response is similarly
-important.
+DUPLICATE_PUSH frame (see {{frame-duplicate-push}}).
+
+Ordering of a PUSH_PROMISE or DUPLICATE_PUSH in relation to certain parts of the
+response is important. The server SHOULD send PUSH_PROMISE or DUPLICATE_PUSH
+frames prior to sending any frames that reference the promised responses.  This
+avoids a race where clients issue requests prior to receiving any PUSH_PROMISE
```

Comment:
This ensures that clients do not issue a request for resources that will be pushed by the server.